xdsl/transforms/experimental/stencil_tensorize_z_dimension.py

```diff
@@ -140,6 +144,24 @@ def arithBinaryOpTensorize(
     rewriter: PatternRewriter,
     /,
 ):
+    def rewrite_scalar_operand(
+        scalar_op: SSAValue, dest_typ: TensorType[Attribute]
+    ) -> SSAValue:
+        if isinstance(scalar_op, OpResult) and isinstance(scalar_op.op, Constant):
+            tens_const = Constant(
+                DenseIntOrFPElementsAttr([dest_typ, ArrayAttr([scalar_op.op.value])])
+            )
+            if len(scalar_op.uses) > 1:
+                rewriter.insert_op(tens_const, InsertPoint.before(scalar_op.op))
+            else:
+                rewriter.replace_op(scalar_op.op, tens_const)
+            return tens_const.result
+        emptyop = EmptyOp((), dest_typ)
+        fillop = FillOp((scalar_op,), (emptyop.tensor,), (dest_typ,))
+        rewriter.insert_op(emptyop, InsertPoint.before(op))
+        rewriter.insert_op(fillop, InsertPoint.before(op))
+        return fillop.res[0]
```

Comment:
one danger with nesting function definitions in Python is that it doesn't act as a namespace, and actually redefines the function every time. It would probably be better to move this function to the top level, and make it private. It would also benefit from a doc comment, as well as the function above it.

Reply:
Thanks, fixed as suggested.